src/commands/schema/pull.ts

```diff
@@ -27,36 +33,50 @@ export default class PullSchemaCommand extends SchemaCommand {
     const { url, secret } = await this.fetchsetup();
 
     try {
-      // Gather remote schema files to download.
-      const filesres = await fetch(new URL("/schema/1/files", url), {
+      // Check if there's a staged schema, and require `--staged` if there is one.
+      const statusres = await fetch(new URL(`/schema/1/staged/status`, url), {
         method: "GET",
         headers: { AUTHORIZATION: `Bearer ${secret}` },
       });
-      const filesjson = await filesres.json();
-      if (filesjson.error) {
-        this.error(filesjson.error.message);
-      }
-
-      // Check if there's a staged schema, and require `--staged` if there is one.
-      const params = new URLSearchParams({
-        version: filesjson.version,
-      });
-      const statusres = await fetch(
-        new URL(`/schema/1/staged/status?${params}`, url),
-        {
-          method: "GET",
-          headers: { AUTHORIZATION: `Bearer ${secret}` },
-        }
-      );
       const statusjson = await statusres.json();
       if (statusjson.error) {
         this.error(statusjson.error.message);
       }
 
-      if (statusjson.status !== "none" && !this.flags?.staged) {
-        this.error("There is a staged schema change. Use --staged to pull it.");
-      } else if (statusjson.status === "none" && this.flags?.staged) {
-        this.error("There are no staged schema changes to pull.");
+      let isStagedPull = false;
+      if (statusjson.status === "none") {
+        if (this.flags?.staged) {
+          this.error("`--staged` does nothing when there is no staged schema.");
+        }
+        if (this.flags?.active) {
+          this.error("`--active` does nothing when there is no staged schema.");
+        }
+      } else {
+        if (this.flags?.staged && this.flags?.active) {
+          this.error(
+            "`--active` and `--staged` cannot be passed at the same time."
+          );
+        } else if (!this.flags?.staged && !this.flags?.active) {
+          this.error(
+            "There is a staged schema. Pick the staged or active schema with `--staged` or `--active`."
+          );
+        } else if (this.flags?.staged) {
+          isStagedPull = true;
```

Comment:
This API is too brittle: No flag combo works unconditionally, they all error out sometimes based on db state.

I think instead you can make `--staged` the default, always pulling the latest schema. The stdout output can communicate clearly what is happening.

Alternatively, (contradicting myself a bit) maybe it's just `--staged` which errors out: `--active` always pulls the active and the default pulls whatever is the latest. That feels a bit inconsistent, though.